gammapy/visualization/utils.py

```diff
@@ -410,3 +410,47 @@ def func_to_fit(x, mu, sigma):
         axe.legend()
 
     return axes, result_list
+
+
+def plot_stat_profile(result, datasets, model_name, ncols=3, fit=None):
+    total_stat = result.total_stat
+
+    if model_name not in datasets.models.names:
+        raise ValueError(
+            f"`model_name` {model_name} not in `datasets.models`. List of valid model name can be access via `datasets.models.names`."
+        )
+
+    fit = fit or Fit()
+
+    model = datasets.models[model_name]
+    free_parameters = model.parameters.free_parameters
+
+    n_plot = len(free_parameters)
+
+    if n_plot == 0:
+        log.warning(f"No free parameters in model {model_name}.")
+        return None
+
+    cols = min(ncols, n_plot)
+    rows = 1 + (n_plot - 1) // cols
+
+    width = 12
+    figsize = (width, width * rows / cols)
+
+    fig, axes = plt.subplots(
+        nrows=rows,
+        ncols=cols,
+        figsize=figsize,
+    )
+
+    for ax, par in zip(axes.flatten(), free_parameters):
+        par.scan_n_values = 17
+        idx = model.parameters.index(par)
+        name = model.parameters_unique_names[idx]
+        profile = fit.stat_profile(datasets=datasets, parameter=par)
+        ax.plot(profile[f"{model.name}.{name}_scan"], profile["stat_scan"] - total_stat)
+        ax.set_xlabel(f"{par.name} [{par.unit}]")
+        ax.set_ylabel("Delta TS")
+        ax.set_title(f"{name}:\n {par.value:.1e} +- {par.error:.1e}")
+
```

Comment:
depending on the dof, should we have horizontal lines corresponding to 1,2 and 3 sigma?